aptos-indexer-processors-sdk/sdk/src/common_steps/version_tracker_step.rs

```diff
@@ -1,76 +1,129 @@
 use crate::{
+    config::{
+        indexer_processor_config::{DbConfig, IndexerProcessorConfig, ProcessorMode},
+        processor_status_saver::{
+            BackfillProgress, BackfillStatus, ProcessorStatus, ProcessorStatusSaver,
+        },
+    },
     traits::{
         pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
     },
     types::transaction_context::TransactionContext,
     utils::errors::ProcessorError,
 };
 use anyhow::Result;
+use aptos_indexer_transaction_stream::utils::parse_timestamp;
 use async_trait::async_trait;
 use std::marker::PhantomData;
 
 pub const DEFAULT_UPDATE_PROCESSOR_STATUS_SECS: u64 = 1;
-/// The `ProcessorStatusSaver` trait object should be implemented in order to save the latest successfully
-/// processed transaction versino to storage. I.e., persisting the `processor_status` to storage.
-#[async_trait]
-pub trait ProcessorStatusSaver {
-    // T represents the transaction type that the processor is tracking.
-    async fn save_processor_status(
-        &self,
-        last_success_batch: &TransactionContext<()>,
-    ) -> Result<(), ProcessorError>;
-}
 
 /// Tracks the versioned processing of sequential transactions, ensuring no gaps
 /// occur between them.
 ///
 /// Important: this step assumes ordered transactions. Please use the `OrederByVersionStep` before this step
 /// if the transactions are not ordered.
-pub struct VersionTrackerStep<T, S>
+pub struct VersionTrackerStep<T, S, D>
 where
     Self: Sized + Send + 'static,
     T: Send + 'static,
-    S: ProcessorStatusSaver + Send + 'static,
+    S: ProcessorStatusSaver + Send + Sync + 'static,
+    D: DbConfig + Send + Sync + Clone + 'static,
 {
     // Last successful batch of sequentially processed transactions. Includes metadata to write to storage.
     last_success_batch: Option<TransactionContext<()>>,
     polling_interval_secs: u64,
     processor_status_saver: S,
+    processor_id: String,
+    processor_config: IndexerProcessorConfig<D>,
     _marker: PhantomData<T>,
 }
 
-impl<T, S> VersionTrackerStep<T, S>
+impl<T, S, D> VersionTrackerStep<T, S, D>
 where
     Self: Sized + Send + 'static,
     T: Send + 'static,
-    S: ProcessorStatusSaver + Send + 'static,
+    S: ProcessorStatusSaver + Send + Sync + 'static,
+    D: DbConfig + Send + Sync + Clone + 'static,
```

Comment:
This doesn't feel right, the processor status saver should entirely encapsulate saving processor status, `DbConfig` feels like a leaky abstraction here. I left some suggestions in other comments for how we can remove it from IndexerProcessorConfig.